### PR TITLE
Turbine code cleanup

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
1. Removed unused vars `our_turf_thermal_conductivity`. It did nothing
2. Removed redundant vars such as
   - `mapped` : It can be replaced with `maploaded` var inside `Initialize()` proc as it simply checks if the turbine was created during round start or not. This does mean that admin spawned turbine parts won't come pre installed with parts but that's not a big deal since they can spawn the parts anyway
   - `has_gas_mix` : All turbine parts set's this var to `TRUE` so might as well make that the default case
   - `gas_theoretical_volume` : This can be passed as an param inside `Initialize()` directly rather than storing it in a var which gets used only once
3. Autodoc for procs, fixed return values of some tool acts & `attackby`(we want to return TRUE to end the attack chain early). Removes balloon alert when opening/closing panel. The visual overlay makes that obvious
4. The turbine now shuts itself off when the room has no power/the turbine is sufficiently damaged. The turbine uses a small amount of power for operation (for that green light, internal electronics & stuff) cause it makes sense 
5. Fixes runtime when opening turbine computer and one of the turbine parts is disconnected
6. Adds Screen tips & fixes examines (they weren't using the `span_notice()` proc) and adds even more examines

## Changelog
:cl:
fix: turbine now shuts itself off when the room apc loses power or if it gets damaged. Also uses a small amount of power for operation of internal electronics, the green light & other stuff
fix: No more runtime in turbine computer when parts are not fully connected
qol: adds screentips & examines for turbine
code: removed unused vars, auto doc procs and cleans up some code in turbine
/:cl:

### CODE REVIEW

**Comment by Pickle-Coding** [comment]
> The turbine now shuts itself off when the room has no power/the turbine is sufficiently damaged. The turbine uses a small amount of power for operation (for that green light, internal electronics & stuff) cause it makes sense

It's a turbine. If gas is moving through the rotors, it's going to spin. That shouldn't need power from the room. Just turn off the lights and controls.

**Comment by Exester509** [comment]
Turbines use their own power for lights and even the compressor. I don't think needing a charged APC makes a lot of sense

**Comment by SyncIt21** [comment]
> It's a turbine. If gas is moving through the rotors, it's going to spin. That shouldn't need power from the room. Just turn off the lights and controls.

That's a good point and it was my idea initially but then we realized it isn't a big deal, just wanted to add some semi realism to the turbine, it's good enough for now

File: code/modules/power/turbine/turbine.dm [hunk 1]
```diff
@@ -196,37 +244,27 @@
...
 
-/**
- * Gets the efficiency of the installed part, returns 0 if no part is installed
- */
+/// Gets the efficiency of the installed part, returns 0 if no part is installed
 /obj/machinery/power/turbine/proc/get_efficiency()
-	if(installed_part)
-		return installed_part.part_efficiency
-	return 0
+	return installed_part ? installed_part.part_efficiency : 0
```
Comment (Ghommie) [inline code/modules/power/turbine/turbine.dm:259]:
```suggestion
	return installed_part?.part_efficiency || 0
```

File: code/modules/power/turbine/turbine_computer.dm [hunk 2]
```diff
@@ -54,20 +54,20 @@
 	var/list/data = list()
 
 	var/obj/machinery/power/turbine/core_rotor/main_control = turbine_core?.resolve()
-
-	data["connected"] = main_control ? TRUE : FALSE
+	data["connected"] =  QDELETED(main_control) ? TRUE : FALSE
```
Comment (Ghommie) [inline code/modules/power/turbine/turbine_computer.dm:57]:
```suggestion
	data["connected"] =  !!QDELETED(main_control)
```

File: code/modules/power/turbine/turbine_computer.dm [hunk 3]
```diff
@@ -54,20 +54,20 @@
...
 	data["rpm"] = main_control.rpm ? main_control.rpm : 0
 	data["power"] = main_control.produced_energy ? main_control.produced_energy : 0
-	data["temp"] = main_control.compressor.input_turf?.air.temperature
 	data["integrity"] = main_control.get_turbine_integrity()
 	data["parts_linked"] = main_control.all_parts_connected
 	data["parts_ready"] = main_control.all_parts_ready()
 
 	data["max_rpm"] = main_control.max_allowed_rpm
 	data["max_temperature"] = main_control.max_allowed_temperature
+	data["temp"] = QDELETED(main_control.compressor) ? 0 : main_control.compressor.input_turf?.air.temperature
```
Comment (Ghommie) [inline code/modules/power/turbine/turbine_computer.dm:70]:
```suggestion
	data["temp"] = main_control.compressor?.input_turf?.air.temperature || 0
```
Provided the `compressor` variable of `main_control` gets properly cleared up on deletion as it should anyway, this is fine.

**Review by Ghommie — APPROVED** [review]
Just a couple binary vs terniary operations stuff (jokes on you `?.` is inheritingly terniary). Tgui num props should be 0 rather than null.

**Pull request merged**